snowpack/src/commands/build.ts

```diff
@@ -26,11 +26,18 @@ import {
   relativeURL,
   removeLeadingSlash,
   replaceExt,
+  hmrFilePath,
 } from '../util';
 import {getInstallTargets, run as installRunner} from './install';
+import { EsmHmrEngine } from '../hmr-server-engine';
 
 const CONCURRENT_WORKERS = require('os').cpus().length;
 
+let hmrEngine: EsmHmrEngine | null = null;
+function getIsHmrEnabled(config: SnowpackConfig) {
+  return config.buildOptions.watch && config.devOptions.hmr;
```

Comment:
So I just checked `config.ts`, and `config.buildOptions.watch` is true by default. This is a bit tricky since we originally talked about this to be opt-in for build.

I'm kind of torn, should we just ship this as opt-in for build? I just think it's so unexpected to be running a separate server during the build command. In that case, we need to remove the default `hmr` value from `DEFAULT_CONFIG` and then handle it as default "on" for dev so that this code you have here is opt-in and runs as expected.

```diff
// in dev.ts
-  const {hmr: isHmr} = config.devOptions;
+  const isHmr = config.devOptions.hmr || true;
```

Reply:
> There's definitely still work to be done here, but I'd rather do it all together in a follow up PR and leave this existing limitation in place for now.

maybe we should print a message to inform user about web socket's address.

Reply:
Yup, that makes sense to me

Reply:
>  In that case, we need to remove the default hmr value from DEFAULT_CONFIG and then handle it as default "on" for dev so that this code you have here is opt-in and runs as expected.

The `config.devOptions.hmr` will be `undefined` and not match our config schema if we don't set the `hmr` value in `loadAndValidateConfig()` function. I prefer to set hmr value in `loadAndValidateConfig()` function according to `dev` or `build`.

Reply:
Our config functions don't know what command they are being run in. 

We give commands the chance to modify the config at startup, but at that point it's too late to tell whether the original option was undefined or true/false.

It's okay to make this optional in the config schema as well, there are several places where we already do this. The current behavior isn't changing, just the `hmr` config value would now support undefined, which lets the different commands handle this case differently.

Reply:
Ok, modified 😁